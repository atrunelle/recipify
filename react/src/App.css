.l-container {
  display: grid;
  grid-template-columns: 40px auto 40px;
  grid-template-rows: 25% 50% auto;
  grid-row-gap: 15px;
  grid-template-areas: "header header header" ". main ." "footer footer footer"; }

main {
  grid-area: main; }
