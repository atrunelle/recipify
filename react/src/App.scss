.app {
  display: flex;
  flex: 1 1 auto;
  backface-visibility: hidden;
  flex-direction: column;
  min-height: 100vh;
  max-width: 100%;
  position: relative;  
}