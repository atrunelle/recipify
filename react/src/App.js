import React, { Component } from 'react';
import { Provider } from 'react-redux';

import './App.css';
import store from './store';

import Header from './components/header/Header';
import Footer from './components/footer/Footer';
import RecipeCreator from './recipe/recipe-creator/RecipeCreator';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="app">
          <Header />
            <main className="l-container">
              <RecipeCreator />
            </main>
            <Footer />
        </div>
      </Provider>
    );
  }
}

export default App;
