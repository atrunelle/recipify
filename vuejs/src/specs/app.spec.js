import { mount } from '@vue/test-utils';
import App from '../app';

describe('Component: app', () => {
  let component;

  beforeEach(() => {
    component = mount(App, {
      stubs: {
        'router-view': '<router-view></router-view>',
        'the-header': '<header></header>',
        'the-alert': '<div></div>',
      },
    });
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
