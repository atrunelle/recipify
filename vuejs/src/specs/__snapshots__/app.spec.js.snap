// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: app should match snapshot 1`] = `
<div data-app="true" class="application theme--light" id="app">
    <div class="application--wrap">
        <div class="layout column justify-space-between">
            <header></header>
            <div class="container grid-list-lg">
                <main>
                    <router-view></router-view>
                </main>
            </div>
            <div></div>
            <footer class="footer grey darken-3" style="height: 32px;">
                <div class="layout row wrap aligh-center justify-center">
                    <div class="flex xs12 text-xs-center white--text">
                        <p class="my-2 body-1">Powered by Vue</p>
                    </div>
                </div>
            </footer>
        </div>
    </div>
</div>
`;
